pkg/ansible/runner.go

```diff
@@ -0,0 +1,208 @@
+package ansible
+
+import (
+	"fmt"
+	"path"
+
+	log "github.com/sirupsen/logrus"
+
+	kbatch "k8s.io/api/batch/v1"
+	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	openshiftAnsibleImage          = "openshift/origin-ansible:v3.7"
+	openshiftAnsibleServiceAccount = "cluster-operator-controller-manager"
+	sshPrivateKeySecret            = "ssh-private-key"
+	operatorNamespace              = "cluster-operator"
+	openshiftAnsibleContainerDir   = "/usr/share/ansible/openshift-ansible/"
+	awsCredentialsSecret           = "aws-credentials"
+
+	// namePrefix is used on configmap and job names. The cluster ID is appended,
+	// followed by a random suffix.
+	namePrefix = "ansible-job-"
+)
+
+type ansibleRunner struct {
+	KubeClient kubernetes.Interface
+	Image      string
+}
+
+func NewAnsibleRunner(kubeClient kubernetes.Interface) *ansibleRunner {
+	return &ansibleRunner{
+		KubeClient: kubeClient,
+		Image:      openshiftAnsibleImage,
+	}
+}
+func (r *ansibleRunner) createInventoryConfigMap(clusterName, inventory, vars string) (*kapi.ConfigMap, error) {
+	log.Debugln("Creating inventory/vars ConfigMap")
+	genNamePrefix := fmt.Sprintf("%s%s-", namePrefix, clusterName)
```

Comment:
I would like to see some additional label added to this name that identifies which specific ansible job is being run. I don't know that the "ansible" portion is necessary. For example, "infra-provision-job-myCluster-" It is fine for now, though.

Reply:
Because we're creating these resources in a single namespace for now, shouldn't the name also include the cluster's namespace?

Reply:
Updated a little to job-infra-clusterid-RANDOM, and now used on both configmap and job exactly.

Reply:
It includes the clusters ID, right now the cluster doesn't have a namespace, I do have a todo on my trello card to test multi-namespace shortly.

Reply:
What do you mean by "the cluster doesn't have a namespace"? The cluster certainly has a namespace. Also, are the terms ID and Name interchangeable in the kube world? Or does the ID of a resource refer to the UID? Or is the term ID not used at all?

Reply:
"ID" is not a kube term of art that I am familiar with. We have metadata.uid and metadata.name.

Reply:
I meant that there isn't a namespace being created per cluster by anything per se, the cluster object itself would land in one when created for sure. I'll send in a follow-up that re-uses that namespace for the configmap and job, which is probably what Cesar meant for me to do.

Cluster ID is coming from my ops rotation, just a prevalent concept in all their ansible, I'll try to remember to stick to Name. Cluster ID on their side is just a short label for each cluster such as "free-int" or "online-stg", well suited for a kube name.